### ISSUE TITLE
Keyboard overlaps app

### ISSUE THREAD

**Issue by frederikheld** [capacitor-community/safe-area#38]
**Describe the bug**

See screenshots.

![keyboard-overlap](https://github.com/user-attachments/assets/3f443750-8e4f-4365-9350-b9f35873c529)

Left: app without keyboard as reference. App bar aligned with bottom of the viewport.

Right: open keyboard overlaps app for an amount of pixels that seems to match the safe area inset bottom (see red squares)

I've tested this with the current `alpha-8`, but it's most probably a regression from this #31.

**Comment by frederikheld** [comment]
So maybe I had my own fix in place that now causes the overlap. I'll look into it.

**Comment by frederikheld** [comment]
The last version that works properly for me is alpha.5. All other versions have issues with either too much or too little safe area inset bottom.

But one thing I noticed with all versions: when I open the keyboard, the bottom bar jumps up for "keyboard height + safe area inset bottom" pixels and a brief moment later jumps back to "keyboard height" (which it should be because the keyboard already covers the bottom bar). The same happens the other way round when closing the keyboard.

I assume this happens because "safe area inset bottom" takes a noticeable amount of time to be updated updated.

**Comment by JanPolasek** [comment]
The reason why it worked in alpha.5 is this commit in alpha.6 https://github.com/capacitor-community/safe-area/commit/acfaa0cda35792e10fd1a67e711af582d633261f
Without `resizeOnFullScreen` IME height already includes the navbar height, so just `imeHeight = imeInsets.bottom` works (at least on my devices and emulators)

With `resizeOnFullScreen` things get a little weird. The view is resized by the Keyboard plugin in a way which doesn't affect the insets that this plugin receives, but it affects the paddings it sets, so not only is it not necessary to subtract navbar height from IME height, it is necessary to *add* it instead. Not quite sure how all of that works, I don't know much about Android development.

I guess if this plugin wanted to support both options it would either need to read the Keyboard plugin's settings, which doesn't sound like a great solution, or somehow find out whether the Keyboard plugin did some resizing and account for that (and I'm not even sure how to go about that)
Alternatively, this plugin could just document that `resizeOnFullScreen` shouldn't be used along with this plugin. Even if it didn't break things, I don't think it would do anything useful either. 

But in both cases there's one more unfortunate bug and that is that custom navbar colours get pushed up 
![Screenshot 2025-01-13 at 14 28 31](https://github.com/user-attachments/assets/a3e7ee03-5f05-4e76-9b9e-e9261272cff2)
While experimenting with that I just set the background to fully transparent when imeHeight > 0 which did work, but that doesn't sound like a great solution either, there's gotta be a way to hide it correctly or make sure it doesn't get affected by the decor view paddings, I just don't know how (yet at least)

**Comment by JanPolasek** [comment]
Alright, the navbar colour issue should either become a separate issue or just be ignored (or rather documented as a known quirk) - it occurs when the resize mode in the Keyboard plugin is set to Native (which happens to be the default), so it's completely unrelated to what's happening here.

**Comment by datawaslost** [comment]
PR looks good, looking forward to it being reviewed and merged into a new version

**Comment by ekrapfl** [comment]
I'd like to add that I am experiencing the issue with the keyboard overlapping the app as well.  I have patched my own app with the fix in #41, but I would love to see a new release with those changes.  Thanks for the fix!

**Comment by miro-ku** [comment]
I'd like to add that using this plugin makes impossible to use Keyboard's plugin setting of `resize: 'none'` which is used in my app, so when the keyboard is shown - the viewport is actually resized, but into a weird size, that includes the bottom inset size, that is hidden behind the keyboard. Is it possible to disable viewport resizing at all?

**Comment by SvenBudak** [comment]
I’m experiencing the same issue. Is there any ETA on when this fix will be released?

**Comment by frederikheld** [comment]
> Capacitor is not using any specific version the way you think I guess. It just uses the system webview, which in turn uses the webview you've installed through Google Play for instance.

Yeah, you're right. I forgot about that. One more argument to not deprecate this plugin. There's tons of old Android versions out there that will never get a fixed WebView and will still be able to run Capacitor apps.

Nobody expects you to continue maintaining this plugin. But keeping it broken as it is creates the false sense of a working solution that will just keep frustrating users.

What would be the issue to just merge the PR and create one last release? Add a note about the upcoming fix and you're not gonna maintain this plugin anymore and everyone will be happy.

**Comment by JanPolasek** [comment]
@frederikheld the easiest option would be to use capacitor's current official solution for the time being which is resizing the webview to essentially nullify edge to edge mode, there's a configuration option for that in the current version.

My PR as it stands is no longer enough to fix things. 

I'm honestly contemplating pushing my current heavily modified version of this library and publishing it on npm because after seeing how Google absolutely fumbled this whole edge to edge thing I have zero trust them and don't expect a real functional implementation anytime soon.
